### DISCUSSION TITLE
Example Client - how to setup for PKCE

### DISCUSSION THREAD

**Discussion by RoxKilly** [ory/hydra#2976] [Q&A] [answered]
I'm going through the [Ory/Hydra 5min tutorial][1].  I am able to create a public Client that will have to login using the authorization_code flow with PKCE like this:

```
    hydra clients create \
    --endpoint http://127.0.0.1:4445 \
    --id public-client \
    --grant-types authorization_code,refresh_token \
    --response-types code \
    --scope offline \
    --token-endpoint-auth-method none
    --callbacks http://127.0.0.1:5555/callback
```

I also configured Hydra to require public clients to use PKCE by setting env variable

```
OAUTH2_PKCE_ENFORCED_FOR_PUBLIC_CLIENTS=true
```

When I follow the tutorial and create the user application like this...

```
    hydra token user \
    --client-id public-client \
    --endpoint http://127.0.0.1:4444/ \
    --port 5555 \
    --scope offline
```

...the application fails to use the PKCE flow:

> invalid_request
>
> This client must include a code_challenge when performing the authorize code flow, but it is missing.

I've looked through the [CLI docs for creating the sample application][2] for the right configuration without success. How do I start the example client application setup to use PKCE?  

  [1]: https://www.ory.sh/hydra/docs/5min-tutorial/
  [2]: https://www.ory.sh/hydra/docs/cli/hydra-token-user/

**Comment by vinckr** [comment, marked as answer]
Hello @RoxKilly 
Apologies for the late answer!
We have a blogpost describing a PKCE setup, maybe that can help you out https://www.ory.sh/oauth2-for-mobile-app-spa-browser/

**Reply by vinckr** [reply]
Hey @RoxKilly , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
